projects/angular-split/src/lib/component/split.component.ts

```diff
@@ -521,7 +523,38 @@ export class SplitComponent implements AfterViewInit, OnDestroy {
     }
   }
 
-  public startDragging(event: MouseEvent | TouchEvent, gutterOrder: number, gutterNum: number): void {
+  public startKeyboardDrag(event: KeyboardEvent, gutterOrder: number, gutterNum: number) {
+    switch (event.key) {
+      case 'ArrowLeft':
+      case 'ArrowRight':
+      case 'ArrowUp':
+      case 'ArrowDown':
+          break
+      default:
+          return
+    }
+
+    if (this.disabled === true || this.isWaitingClear === true) {
+      return
+    }
+
+    const endPoint = getKeyboardEndpoint(event, this.direction)
+    if (endPoint === null) {
+      return
+    }
+    this.endPoint = endPoint
+    this.startPoint = getPointFromEvent(event)
+
+    event.preventDefault()
+    event.stopPropagation()
+
+    this.setupForDragEvent(gutterOrder, gutterNum)
+    this.startDragging()
+    this.drag()
+    this.stopDragging()
```

Comment:
Comment from internal team member -
Validate for yourself in the code that this can handle negative coordinates and coordinates that are wider/taller

Reply:
There is logic in drag() to calculate and "absorb" extra offset calculated in areasBefore/After.remain

From a quick look 'updateAreaSize' does not act upon the "pixelRemain" value in any way, discarding the excess, while using the "pixelAbsorb". Here is what it looks like if a gutter 6px from the bottom is keyed down:
```
percentAfterAbsorption: 0
pixelAbsorb: -6
pixelRemain: -44
```